python/plugins/processing/algs/qgis/CreateAtlasGrid.py

```diff
@@ -0,0 +1,178 @@
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+
+        layout_param = QgsProcessingParameterString(
+            self.LAYOUT,
+            description=self.tr('Layout to use'),
+            defaultValue=None,
+            optional=False)
+
+        layout_param.setMetadata(
+            {'widget_wrapper': {
+                'class':
+                    'processing.gui.wrappers_layout.LayoutWrapper'}})
+        self.addParameter(layout_param)
+
+        self.addParameter(QgsProcessingParameterFeatureSource(self.COVERAGE_LAYER,
+                                                              self.tr('Coverage layer'),
+                                                              [QgsProcessing.TypeMapLayer]))
+        self.addParameter(QgsProcessingParameterDistance(self.COVERAGE_BUFFER,
+                                                         self.tr('Coverage Buffer'), parentParameterName='COVERAGE_LAYER',
+                                                         defaultValue=0.0))
+        self.addParameter(QgsProcessingParameterBoolean(self.ONLY_ON_FEATURES,
+                                                        self.tr('Only on intersection features'),
+                                                        defaultValue=True))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Grid'),
+                                                            QgsProcessing.TypeVectorPolygon))
+
+    def processAlgorithm(self, parameters, context, feedback):
+
+        layout_name = self.parameterAsString(
+            parameters,
+            self.LAYOUT,
+            context
+        )
+        coverage_layer = self.parameterAsVectorLayer(
+            parameters,
+            self.COVERAGE_LAYER,
+            context
+        )
+        coverage_distance = self.parameterAsDouble(
+            parameters,
+            self.COVERAGE_BUFFER,
+            context
+        )
+        only_on_features = self.parameterAsBool(
+            parameters,
+            self.ONLY_ON_FEATURES,
+            context
+        )
+
+        fields = QgsFields()
+        fields.append(QgsField('id', QVariant.Int, '', 10, 0))
+        (sink, dest_id) = self.parameterAsSink(
+            parameters,
+            self.OUTPUT,
+            context,
+            fields,
+            QgsWkbTypes.Polygon,
+            coverage_layer.sourceCrs()
+        )
+
+        if sink is None:
+            raise QgsProcessingException(self.invalidSinkError(parameters, self.OUTPUT))
+
+        layout = QgsProject().instance().layoutManager().layoutByName(layout_name)
+        map = layout.referenceMap()
+        h = map.extent().height()
+        w = map.extent().width()
+
+        coverage = coverage_layer.sourceExtent().buffered(coverage_distance)
+
+        xmin = coverage.xMinimum()
+        xmax = coverage.xMaximum()
+        resto = (w - ((xmax - xmin) % w)) / 2
```

Comment:
```suggestion
        remainder = (w - ((xmax - xmin) % w)) / 2
```

as in https://en.wikipedia.org/wiki/Remaind